library-backend/library-app/src/main/java/org/internship/library/app/controller/BookController.java

```diff
@@ -27,45 +27,63 @@
 
     /**
      * Retrieves the book specified by the id
+     *
      * @param id the book id
-     * @return the book
+     * @return the book OR NOT_FOUND(404)
      */
     @GetMapping("/{id}")
     public ResponseEntity<Book> getBook(@PathVariable String id) {
-        return ResponseEntity.status(HttpStatus.OK).body(bookService.getBook(id));
-   }
+        logger.info("Retrieving book with the id of: " + id);
+
+        try {
+            return ResponseEntity.status(HttpStatus.OK).body(bookService.getBook(id));
+        } catch (Exception e) {
```

Comment:
Do you return 404 for ANY exception?

What if, for example, the book exists but there is a timeout? is the 404 correct?

Reply:
Same for all the other methods.